### PR TITLE
Update FileDropZone's documentation

### PR DESCRIPTION
**Pull request by Fercas123**
- updating documentation for Filedropzone from feedback

### CODE REVIEW

File: site/docs/components/file-drop-zone/usage.mdx [hunk 1]
```diff
@@ -11,9 +11,23 @@ data:
 
 ### Using the component
 
-Make sure you communicate any limits, such as maximum file size and type restrictions, before the user uploads their file. If the upload fails, clearly explain the reason why and how to resolve it.
+#### Composition
+
+When composing the fileDropzone, consider using `FileDropZoneIcon` to help inform users by showing an upload icon while items have not been updated and a status icon giving feedback about the state of the input.
```
Comment (joshwooding) [inline site/docs/components/file-drop-zone/usage.mdx:16]:
```suggestion
When composing the file drop zone, consider using `FileDropZoneIcon` to help inform users by showing an upload icon while items have not been updated and a status icon giving feedback about the state of the input.
```

File: site/docs/components/file-drop-zone/usage.mdx [hunk 2]
```diff
@@ -11,9 +11,23 @@ data:
...
+It is recommended to pass a title paragraph to the component to indicate the expected action.
+You can customize the `FileDropZoneTrigger`'s text by passing `children` into the component.
+If specific criteria is required, make sure you communicate any limits, such as maximum file size and type restrictions as validation text, before the user uploads their file. If the upload fails, clearly explain the reason why and how to resolve it.
+
+####     Validation and acceptance criteria
```
Comment (joshwooding) [inline site/docs/components/file-drop-zone/usage.mdx:21]:
```suggestion
#### Validation and acceptance criteria
```

File: site/docs/components/file-drop-zone/usage.mdx [hunk 3]
```diff
@@ -11,9 +11,23 @@ data:
 
 ### Using the component
 
-Make sure you communicate any limits, such as maximum file size and type restrictions, before the user uploads their file. If the upload fails, clearly explain the reason why and how to resolve it.
+#### Composition
+
+When composing the file drop zone, consider using `FileDropZoneIcon` to help inform users by showing an upload icon while items have not been updated and a status icon giving feedback about the state of the input.
+It is recommended to pass a title paragraph to the component to indicate the expected action.
```
[the remark this answers is not in the document]
Comment (joshwooding) [reply site/docs/components/file-drop-zone/usage.mdx]: I've unresolved this chat as I don't think it was addressed.

[the remark this answers is not in the document]
Comment (joshwooding) [reply site/docs/components/file-drop-zone/usage.mdx]: What is a "title paragraph"?

File: site/docs/components/file-drop-zone/usage.mdx [hunk 4]
```diff
@@ -11,10 +11,22 @@ data:
...
+- Communicate any requirements, such as maximum file size and type restrictions as validation text, before the user uploads their file. If the upload fails, clearly explain the reason why and how to resolve it.
+
+#### Validation and acceptance criteria
+
+- You can specify acceptable file types with `FileDropZoneTrigger`'s `accept` prop. The prop doesn't validate the selected files' types. Instead, it provides the acceptable types to the browser, which in turn can guide users towards selecting the correct file types.
+- Make sure that the acceptance criteria is backed up by appropriate validation. Use `FileDropZone`'s `onDrop` prop to validate the file types.
 
 Inform users when files successfully upload to reassure them that the action has completed.
 
+- To inform users of the criteria, you can display custom validation text on the file drop zone. If uploaded files fail to meet the validation criteria, the file drop zone will display an error state.
```
Comment (origami-z) [inline site/docs/components/file-drop-zone/usage.mdx:28]: "will display" sounds the component will do it itself? but rather we're asking the user to show the error state?

File: site/docs/components/file-drop-zone/usage.mdx [hunk 5]
```diff
@@ -11,10 +11,22 @@ data:
...
 
-Make sure you communicate any limits, such as maximum file size and type restrictions, before the user uploads their file. If the upload fails, clearly explain the reason why and how to resolve it.
+#### Composition
+
+- When composing the file drop zone, consider using `FileDropZoneIcon` to inform users about the current status. You can show an upload icon when the user has not yet uploaded an item, or a status icon that communicates the current upload status.
+- We recommend passing a header to the component to indicate the expected action, such as "drop files here."
+- You can customize the `FileDropZoneTrigger`'s text by passing `children` into the component.
+- Communicate any requirements, such as maximum file size and type restrictions as validation text, before the user uploads their file. If the upload fails, clearly explain the reason why and how to resolve it.
+
+#### Validation and acceptance criteria
```
Comment (origami-z) [inline site/docs/components/file-drop-zone/usage.mdx:21]: "validation" is a concept only appearing in "how to" tab. I don't think we have anything in the examples tab associated with this. e.g. are we talking about text below the trigger button should be validation message?

File: site/docs/components/file-drop-zone/usage.mdx (hunk 4 above)
Comment (Fercas123) [inline site/docs/components/file-drop-zone/usage.mdx:28]: @mikearildbrown does this sound right?
```suggestion
- To inform users of the criteria, you can display custom validation text on the file drop zone. If uploaded files fail to meet the validation criteria, change `state` prop to `error` so the file drop zone will display the error state.
```

Comment (Fercas123) [reply site/docs/components/file-drop-zone/usage.mdx]: the state would change the color of the box to red, if the user wants to, they can also extend the text within the component to further explain the error

File: site/docs/components/file-drop-zone/examples.mdx [hunk 6]
```diff
@@ -15,21 +15,26 @@ data:
...
+    We recommend passing a header to the component to indicate the expected action, such as "Drop files here".
+    Communicate any requirements, such as maximum file size and type restrictions as validation text within the component.
 
   </LivePreview>
   <LivePreview componentName="file-drop-zone" exampleName="Error">
 
     ### Error
 
     Inform users when a file cannot be uploaded successfully. Error messages should provide feedback to guide the user resolve the issue.
+    When uploaded files fail to meet the validation criteria, change `state` prop to `error` so the file drop zone will display the error state.
```
Comment (joshwooding) [inline site/docs/components/file-drop-zone/examples.mdx:29]:
```suggestion
    When uploaded files fail to meet the validation criteria, change `state` prop to `"error"` so the file drop zone will display the error state.
```

File: site/docs/components/file-drop-zone/examples.mdx [hunk 7]
```diff
@@ -15,21 +15,26 @@ data:
...
 
   </LivePreview>
   <LivePreview componentName="file-drop-zone" exampleName="Success">
 
     ### Success
 
     Inform users when files successfully upload. Success messages should reassure users that the action has been completed.
+    When files uploaded successfully, change `state` prop to `success` so the file drop zone will display the success state.
```
Comment (joshwooding) [inline site/docs/components/file-drop-zone/examples.mdx:37]:
```suggestion
    When files uploaded successfully, change `state` prop to `"success"` so the file drop zone will display the success state.
```

File: site/docs/components/file-drop-zone/usage.mdx [hunk 8]
```diff
@@ -11,10 +11,22 @@ data:
...
+
+- You can specify acceptable file types with `FileDropZoneTrigger`'s `accept` prop. The prop doesn't validate the selected files' types. Instead, it provides the acceptable types to the browser, which in turn can guide users towards selecting the correct file types.
+- Make sure that the acceptance criteria is backed up by appropriate validation. Use `FileDropZone`'s `onDrop` prop to validate the file types.
 
 Inform users when files successfully upload to reassure them that the action has completed.
 
+- To inform users of the criteria, you can display custom validation text on the file drop zone. When uploaded files fail to meet the validation criteria, change the `state` prop to `error` so the file drop zone displays the error state.
```
Comment (joshwooding) [inline site/docs/components/file-drop-zone/usage.mdx:28]:
```suggestion
- To inform users of the criteria, you can display custom validation text on the file drop zone. When uploaded files fail to meet the validation criteria, change the `state` prop to `"error"` so the file drop zone displays the error state.
```

File: site/docs/components/file-drop-zone/usage.mdx (hunk 5 above)
Comment (Fercas123) [reply site/docs/components/file-drop-zone/usage.mdx:21]: added some more validation context within the examples

**Review by joshwooding — APPROVED** [review]

**Review by feliciackh — APPROVED** [review]

**Pull request merged**